packages/commonwealth/client/scripts/views/pages/CommunityManagement/AdminsAndModerators/ManageRoles/ManageRoles.tsx

```diff
@@ -23,39 +23,20 @@ export const ManageRoles = ({
 }: ManageRoleRowProps) => {
   const navigate = useCommonNavigate();
 
-  const removeRole = async (role: RoleInfo) => {
-    try {
-      const res = await axios.post(`${app.serverUrl()}/upgradeMember`, {
-        community_id: app.activeChainId(),
-        new_role: 'member',
-        address: role.Address.address,
-        jwt: app.user.jwt,
-      });
+  let roleToBeDeleted;
+  const { useRemoveRolesMutation } = updateRoles;
+  const { mutateAsync: removeRole } = useRemoveRolesMutation({
+    onRoleUpdate,
+    roleToBeDeleted,
+  });
 
-      if (res.data.status !== 'Success') {
-        throw new Error(`Got unsuccessful status: ${res.data.status}`);
-      }
-
-      const roleData = res.data.result;
-      const newRole = new RoleInfo({
-        id: roleData.id,
-        address_id: roleData.address_id,
-        address_chain: roleData.community_id,
-        address: roleData.address,
-        community_id: roleData.community_id,
-        permission: roleData.permission,
-        allow: roleData.allow,
-        deny: roleData.deny,
-        is_user_default: roleData.is_user_default,
-      });
-      onRoleUpdate(role, newRole);
-    } catch (err) {
-      const errMsg = err.response?.data?.error || 'Failed to alter role.';
-      notifyError(errMsg);
-    }
-  };
+  const { data: { admins: returnedAdmins, mods: returnedMods } = {} } =
+    useFetchAdminQuery({
+      communityId: app.activeChainId(),
+    });
 
   const handleDeleteRole = async (role: RoleInfo) => {
+    roleToBeDeleted = role;
```

Comment:
please do not use global variable, what is the reason to keep `roleToBeDeleted` outside of the function?